airbyte-cdk/bulk/toolkits/extract-jdbc/src/main/kotlin/io/airbyte/cdk/output/JdbcExceptionClassifier.kt

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.output
+
+import io.airbyte.cdk.Operation
+import io.micronaut.context.annotation.Replaces
+import io.micronaut.context.annotation.Value
+import java.sql.SQLException
+
+@Replaces(ExceptionClassifier::class)
+class JdbcExceptionClassifier(@Value("\${${Operation.PROPERTY}}") operationName: String) :
```

Comment:
Would extending this class in the connectors be the way to implement the error framework that @theyueli added  e.g.  `PostgresSourceExceptionHandler`?

Reply:
Yes and no. This class such as it is exists only to maintain existing functionality, which is pretty bare bones. I didn't want to add too much stuff to this PR. However, it's not really what we want; what we want is a bunch of classifiers among which @theyueli 's (the one with `ConnectorErrorProfile`s) would be one of them.

I see adding functionality as the logical next step. Micronaut really unlocks a lot of possibilities here, it should be trivial to add some kind of rule definitions into a yaml file such as https://github.com/airbytehq/airbyte-enterprise/blob/master/airbyte-integrations/connectors/source-firetruck/src/main/resources/application.yml

The obvious first choice is to have some kind of rules engine that works on SQLException vendor codes.